tool/tctl/common/auth_command.go

```diff
@@ -748,6 +754,53 @@ func (a *AuthCommand) checkProxyAddr(clusterAPI auth.ClientI) error {
 	return trace.BadParameter("couldn't find registered public proxies, specify --proxy when using --format=%q", identityfile.FormatKubernetes)
 }
 
+func (a *AuthCommand) logUngrantedTTL(clusterAPI auth.ClientI) error {
+	rc := clockwork.NewRealClock()
+	var addr string
+	var u *utils.NetAddr
+	if a.proxyAddr != "" {
+		var err error
+		u, err = utils.ParseAddr(a.proxyAddr)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		addr = u.Addr
+	}
+	status, err := client.StatusFor("", addr, a.genUser)
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
+	if status != nil && !status.IsExpired(rc) {
+		remaining := status.ValidUntil.Sub(rc.Now())
+		if a.genTTL > remaining {
+			logrus.Warnf("requested ttl of %s cannot be granted. "+
+				"Existing session has a TTL that ends in %s. proceeding with %s ttl.",
+				a.genTTL, remaining, remaining,
+			)
+		}
+		return nil
+	}
+	user, err := clusterAPI.GetUser(a.genUser, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, userRole := range user.GetRoles() {
+		role, err := clusterAPI.GetRole(context.TODO(), userRole)
```

Comment:
Ditch context.TODO and pass the context as a parameter. I realize it won't go much far, but it's still better context discipline.